cloudlink-combined-storage/src/main/java/com/gluonhq/combinedstorage/service/Service.java

```diff
@@ -0,0 +1,126 @@
+ * notice, this list of conditions and the following disclaimer in the
+ * documentation and/or other materials provided with the distribution.
+ *     * Neither the name of Gluon, any associated website, nor the
+ * names of its contributors may be used to endorse or promote products
+ * derived from this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL GLUON BE LIABLE FOR ANY
+ * DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package com.gluonhq.combinedstorage.service;
+
+import com.gluonhq.cloudlink.client.data.DataClient;
+import com.gluonhq.cloudlink.client.data.DataClientBuilder;
+import com.gluonhq.cloudlink.client.data.OperationMode;
+import com.gluonhq.cloudlink.client.data.SyncFlag;
+import com.gluonhq.connect.GluonObservableList;
+import com.gluonhq.connect.provider.DataProvider;
+import com.gluonhq.combinedstorage.model.Note;
+import java.util.ArrayList;
+import java.util.List;
+import javafx.beans.binding.Bindings;
+import javax.annotation.PostConstruct;
+
+public class Service {
+    
+    private static final String NOTES = "notes-combined";
+    
+    /**
+     * True uses cloud storage with preference over local storage
+     * False disables cloud storage
+     */
+    private static final boolean CLOUD_STORAGE_ENABLED = true;
+
+    private GluonObservableList<Note> notes;
+    
+    private DataClient localDataClient;
+    private DataClient cloudDataClient;
+    
+    @PostConstruct
+    public void postConstruct() {
+        localDataClient = DataClientBuilder.create()
+                .operationMode(OperationMode.LOCAL_ONLY)
+                .build();
+        
+        cloudDataClient = DataClientBuilder.create()
+                .operationMode(OperationMode.CLOUD_FIRST)
+                .build();
+        
+        notes = retrieveNotes();
+    }
+    
+    private GluonObservableList<Note> retrieveNotes() {
+        // Read local notes first
+        GluonObservableList<Note> internalLocalNotes = DataProvider.retrieveList(
+                localDataClient.createListDataReader(NOTES, Note.class,
+                            SyncFlag.LIST_WRITE_THROUGH,
+                            SyncFlag.OBJECT_WRITE_THROUGH));
+
+        internalLocalNotes.initializedProperty().addListener((obs, ov, nv) -> {
```

Comment:
Without GCL credentials, I get this exception:
```
[SUB] Exception in thread "GluonCloudListDataReaderImpl#notes-combined" javax.json.stream.JsonParsingException: Unexpected char 70 at (line no=1, column no=1, offset=0)
[SUB] 	at org.glassfish.json.JsonTokenizer.unexpectedChar(JsonTokenizer.java:577)
[SUB] 	at org.glassfish.json.JsonTokenizer.nextToken(JsonTokenizer.java:394)
[SUB] 	at org.glassfish.json.JsonParserImpl$NoneContext.getNextEvent(JsonParserImpl.java:402)
[SUB] 	at org.glassfish.json.JsonParserImpl.next(JsonParserImpl.java:352)
[SUB] 	at org.glassfish.json.JsonReaderImpl.readObject(JsonReaderImpl.java:87)
[SUB] 	at com.gluonhq.impl.cloudlink.client.data.GluonCloudListDataReaderImpl.lambda$finalIterator$4(GluonCloudListDataReaderImpl.java:220)
[SUB] 	at java.lang.Thread.run(Thread.java:834)
[SUB] 	at com.oracle.svm.core.thread.JavaThreads.threadStartRoutine(JavaThreads.java:515)
[SUB] 	at com.oracle.svm.core.posix.thread.PosixJavaThreads.pthreadStartRoutine(PosixJavaThreads.java:193)
```

Reply:
This has been fixed with UP-105

Reply:
This has been fixed with UP-105